tests/environment.py

```diff
@@ -336,3 +349,83 @@ def launch_environment_docker(unique_id: str, skip_init: bool = False, tag: str
     debug('agent2: {}'.format(agents[1].mac))
     debug('agent2 wlan0: {}'.format(agents[1].radios[0].mac))
     debug('agent2 wlan2: {}'.format(agents[1].radios[1].mac))
+
+
+class ALEntityPrplWrt(ALEntity):
+    """Abstraction of ALEntity in real device."""
+
+    def __init__(self, name: str, device: None, is_controller: bool = False):
+        self.device = device
+        self.name = name
+        self.bridge_name = 'br-lan'
+
+        if is_controller:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_controller.conf'
+        else:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_agent.conf'
+
+        ucc_port = self.command(("grep \"ucc_listener_port\" {} "
+                                "| cut -d'=' -f2 | cut -d\" \" -f 1").format(self.config_file_name))
+
+        device_ip_output = self.command(
+                "ip -f inet addr show {} | head -n 2".format(self.bridge_name))
+        device_ip = re.search(
+            r'inet (?P<ip>[0-9.]+)', device_ip_output.decode('utf-8')).group('ip')
```

Comment:
Regular expressions are better be explained explicitly. They are not part of the `common` or `expected` knowledge from the average maintainer. Moreover, there are few dialects of regexp so even if one is familiar with regexp she would still need to look for the specific one (python in this case).